City-of-Helsinki/drupal-helfi-rekry
# UHF-8015 & UHF-8027: Rekry migration and search app fixes

# [UHF-8015](https://helsinkisolutionoffice.atlassian.net/browse/UHF-8015) & [UHF-8027](https://helsinkisolutionoffice.atlassian.net/browse/UHF-8027)

## What was done
* Fixes for task area migration to help with language versions
* Fix employment type filter with form reset functionality
* Fix youth summer jobs filter visibility
* Fix search results label translations
* Improved handling of 403 redirect logic

## Install & replicate issues
**Setup for ES proxy and migrations:**
* `git checkout dev`
* `make up`
* Run `make ps` to get the correct proxy port: `drupal-helfi-rekry-elastic`. Should be 9200 but might change if you have multiple ES proxies running.
* Edit the `.env` file:
  * Change the port in `ELASTIC_PROXY_URL`
  * Add the correct client `HELBIT_CLIENT_ID` value. You can get this from Rekry's Azure Library (same ID is used on all environments)
* Run `make up` to update the environment values
**Make sure your instance is up-to-date:**
* `make fresh`
* If you get an `ERROR 1273 (HY000) at line 14784: Unknown collation: 'utf8mb4_0900_ai_ci'` error with the database import, run `make shell` and then:
  * `sed -i 's/utf8mb4_0900_ai_ci/utf8mb4_swedish_ci/g' dump.sql; drush sql-query --file=/app/dump.sql && drush helfi:oc:sanitize-database && drush cim -y;drush updb -y;drush cr;drush uli` 
* Run these in the container:
```
drush migrate-import helfi_rekry_task_areas:all --update
drush migrate-import helfi_rekry_task_areas:all_en --update
drush migrate-import helfi_rekry_task_areas:all_sv --update
drush migrate-import helfi_rekry_jobs:all --update
drush migrate-import helfi_rekry_jobs:all_sv --update
drush migrate-import helfi_rekry_jobs:all_en --update
drush cron
drush sapi-rt
drush sapi-i
```

**Replicate issues:**
* [x] On the search page, make some selections from the "Palvelussuhteen tyyppi / Type of employment relationship" dropdown and click search
  * The filter should work and the selections should appear as buttons, but the "Clear selections" option should not appear
  * Add some other filters. Now the "Clear selections" option should be visible, but when you click on it it shouldn't get rid of the employment relationship selections.
* [x] The "Nuorten kesätyöt / Youth summer jobs" checkbox should be missing
* [x] Open the search listing on swedish and english versions. The result stats should just say "xx results" instead of "xx results (yy listings)" when you compare to the finnish version
* [ ] Go to https://helfi-rekry.docker.so/fi/avoimet-tyopaikat/admin/structure/taxonomy/manage/task_area/overview and find the swedish translation for "Varhaiskasvatuksen lastenhoitajat". When you edit it, the title should have some whitespace at the end of the term name
* [x] Go to https://helfi-rekry.docker.so/fi/avoimet-tyopaikat/admin/content?title=&type=All&status=2&langcode=All and open an unpublished job listing without being logged in. You should be redirected to a 404 page with the URL: https://helfi-rekry.docker.so/job-listing-removed
 
## How to install fixes
* `git checkout UHF-8015-rekry-migration-translation-fixes`
* Run `make drush-cim drush-cr`
* Run the following commands in the container:
```
drush locale:check
drush locale:update
drush migrate-import helfi_rekry_task_areas:all_sv --update
drush migrate-import helfi_rekry_jobs:all --update
drush migrate-import helfi_rekry_jobs:all_sv --update
drush sapi-rt
drush sapi-i
```

## How to test
* [x] The "Clear selection" functionality should now work with the "Palvelussuhteen tyyppi / Type of employment relationship" filter
* [x] The "Nuorten kesätyöt / Youth summer jobs" checkbox should be visible
* [x] Open the search listing on swedish and english versions. The results counter should now be in the format "xx results (yy listings)" and translated correctly on all languages
  * If they aren't translated try going to https://helfi-rekry.docker.so/fi/avoimet-tyopaikat/admin/reports/translations and / or running `make drush-cr` a few times
* [x] Reload the "Varhaiskasvatuksen lastenhoitajat" term's swedish translation. It should no longer have extra whitespace in the title
* [x] Go to https://helfi-rekry.docker.so/fi/avoimet-tyopaikat/admin/tools/rekry-content. There should be a new field titled "Closed job listing redirect page"
  * Use the autocomplete field to search for a landing page titled "Ilmoituksen hakuaika on päättynyt" (this should come with the test site database, if not, use some other node).
  * Try opening unpublished job listings on all languages. You should be redirected to the correct language version of the landing page
  * Run `make drush-cim` to make sure the setting is config ignored properly and doesn't get overridden by config imports

## Designers review


[UHF-8015]: https://helsinkisolutionoffice.atlassian.net/browse/UHF-8015?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[UHF-8027]: https://helsinkisolutionoffice.atlassian.net/browse/UHF-8027?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
